### PR TITLE
Fix Camera Capture on Retake and Close

### LINKED ISSUE #9058: Release the Mic in file upload component after the use
**Describe the bug**

Post recording the audio in the file upload component, the platform is showing the mic is still in use 

**To Reproduce**

1. Go to patient tab
2. Click on any active consultation patient
3. go to files section and record a audio
4. post recording
6. See error

**Expected behavior**

- [ ] mic should be released in file upload component after the use

**Screenshots**

Image

### LINKED ISSUE #9179: Captured Image Saved Unnecessarily When Clicking Retake in Camera Capture
### **Bug Description**
When clicking "Retake" in the camera capture section of the patient file upload, the captured file is unnecessarily saved.

---

### **Steps to Reproduce**
1. Navigate to `/patients`.
2. Open any patient's details.
3. Scroll down to the "Upload Files" section and click on it.
4. Select "Upload Camera."
5. Click "Capture" and then "Retake."
6. Close the capture modal.
7. **Error:** The image is saved even though the user clicked "Retake."

---

### **Expected Behavior**
The image should not be saved when the user clicks "Retake." Only the final captured image should be stored.

---

### **Screenshots**
Image

---

### **Environment**
- **OS:** Ubuntu  
- **Browser:** Chrome  
- **Version:** 131.0.6778.85

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes  

- Fixes #9179  
- Added `clearFiles` function to reset file states (`setFiles([])` and `setUploadFileNames([])`) when "Retake or Close" is clicked.  
- Ensured no files are saved when the modal is closed after clicking "Retake or Close".  

@ohcnetwork/care-fe-code-reviewers  

---

## Merge Checklist  

- [ ] Add specs that demonstrate bug / test a new feature.  
- [ ] Update [product documentation](https://docs.ohc.network) to reflect the changes. 
- [x] Ensure that UI text is kept in I18n files.  
- [ ] Prep screenshot or demo video for changelog entry, and attach it to the issue.  
- [ ] Request for Peer Reviews.  
- [ ] Completion of QA.  

## Summary by CodeRabbit

- **New Features**
	- Introduced a reset functionality in the Camera Capture dialog, allowing users to clear their captured images and uploaded files.
	- Enhanced audio recording management in the Audio Capture dialog with improved microphone access and resource handling.
- **Bug Fixes**
	- Improved handling of the reset action after capturing an image through updated button interactions.
- **Refactor**
	- Enhanced clarity and reusability of the file clearing process in the file upload functionality.

### CODE REVIEW

File: src/components/Files/CameraCaptureDialog.tsx [hunk 1]
```diff
@@ -15,10 +15,11 @@ export interface CameraCaptureDialogProps {
   show: boolean;
   onHide: () => void;
   onCapture: (file: File, fileName: string) => void;
+  onResetCapture: () => void;
```
[the remark this answers is not in the document]
Comment (JavidSumra) [reply src/components/Files/CameraCaptureDialog.tsx:18]: Already Added Prop Types in CameraCaptureDialog Component

**Comment by JavidSumra** [comment]
Hey @Jacobjeevan  can you please review this PR

File: src/components/Files/AudioCaptureDialog.tsx [hunk 2]
```diff
@@ -57,6 +59,7 @@ export default function AudioCaptureDialog(props: AudioCaptureDialogProps) {
 
   const handleStopRecording = () => {
     if (status !== "RECORDING") return;
+    // releaseMic();
```
Comment (Jacobjeevan) [inline src/components/Files/AudioCaptureDialog.tsx]: Is this commented unintentionally?

Comment (JavidSumra) [reply src/components/Files/AudioCaptureDialog.tsx]: Yup I'll remove it sorry 😔

**Comment by nihal467** [comment]
LGTM

**Review by Jacobjeevan — APPROVED** [review]
LGTM

**Review by shivankacker — APPROVED** [review]

File: src/components/Files/AudioCaptureDialog.tsx [hunk 3]
```diff
@@ -101,10 +103,21 @@ export default function AudioCaptureDialog(props: AudioCaptureDialogProps) {
...
+  };
+
   useEffect(() => {
     if (autoRecord && show && status === "RECORDING") {
       handleStartRecording();
     }
+
+    return () => {
+      mediaStreamRef.current && releaseMic();
+    };
```
Comment (rithviknishad) [inline src/components/Files/AudioCaptureDialog.tsx]: line 107 already checks if `mediaStreamRef.current` is truthy or not. is redundant checks necessary?

Also why not just simply inline the releaseMic function here?
```suggestion
    return () => {
      if (mediaStreamRef.current) {
        mediaStreamRef.current.getTracks().forEach((track) => track.stop());
        mediaStreamRef.current = null;
      }
    };
```

**Pull request merged**